### PR TITLE
Restructure the Sphinx doc

### PR DESCRIPTION
**Pull request by flrnslbch**
Habe hiermit mal einen Aufschlag gemacht. Da fehlt auf jeden Fall noch Content :)

## Summary by CodeRabbit

- **Documentation Updates**
  - Updated copyright year and project version settings to adjust dynamically.
  - Enhanced HTML documentation output with new logo and version display.
  - Added a FAQ section explaining the name "Molara."
  - Provided detailed user installation instructions for the Molara package.
  - Introduced a new developer guide for setting up the Molara development environment.
  - Updated README with instructions for generating documentation and starting Molara.
- **New Features**
  - Added new optional and development dependencies for documentation generation.
- **Refactor**
  - Renamed `CONTRIBUTING.md` to `Contribution Guidelines` to clarify its purpose.

### CODE REVIEW

**Comment by adrianusler** [comment]
Linting problems are now fixed @flrnslbch @Michel-Heinz

**Comment by adrianusler** [comment]
Is this ready to merge @flrnslbch?

File: docs/source/users/faq.rst [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Frequently Asked Questions
+==========================
+
+**What does the name Molara mean?**
+
+Molara is a made up of "Mol" -- short for molecule -- and "Ara" which refers to a genus of colorful birds, symbolizing the program's aim to bring vibrant and dynamic visualization capabilities to the study of molecules. Together, "Molara" embodies the fusion of scientific utility with visual elegance, underscoring its purpose as a tool for making the complex and intricate world of molecules as accessible and engaging as the vivid beauty of Ara birds. The section about what the name symbolizes may have been written by ChatGPT but sounds less boring than the actual explanation.
+
+**Where do babys come from?**
```
Comment (adrianusler) [inline docs/source/users/faq.rst:8]: Do we really want to have this in the documentation?

**Comment by adrianusler** [comment]
The docs look really nice, @flrnslbch! :+1:
Like I wrote in my previous comment, the joke about FAQs does not fully convince me. Otherwise, I think this is ready to merge, @Michel-Heinz!

**Review by adrianusler — APPROVED** [review]

File: docs/source/users/install.rst [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Installation
+==============
+
+...
```
[the remark this answers is not in the document]
Comment (ab5424) [reply docs/source/users/install.rst:4]: Yes

File: docs/source/dev/install.rst [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Installation (Dev Version)
+==========================================
+
+...
```
[the remark this answers is not in the document]
Comment (ab5424) [reply docs/source/dev/install.rst:4]: Yes

File: docs/source/users/usage.rst [hunk 4]
```diff
@@ -0,0 +1,9 @@
+Usage
+========
+
+Visualization
+--------------
+
+Molara is able to visualize the output geometries of several quantum chemistry
+programs software packages.
+...
```
[the remark this answers is not in the document]
Comment (ab5424) [reply docs/source/users/usage.rst:9]: Yes

**Pull request merged**